Run matrix profile to the left and to the right

jrecuerda opened the pull request:
A new flag(_left_right_) has been added to the OperationInfo structure
 which allows to reduce the indexes to the left and to the right when
 running self_join. If the flag is set to true the matrix profile to
 the left will be returned in profile_a and the matrix profile to the
 right in profile_b.

zpzim commented:
This does not build! Please revert the changes to the profile array access pattern. profile.data[0] (or profile.data.front()) is necessary because data can have potentially more than one element.

jrecuerda commented:
I moved my code from a dirty branch to a cleaner one and I missed the main.cpp. As you mention I have reverted the change of ProfileData from a vector to a single instance.

jrecuerda commented:
My intention was to add a more higher level meaning, but you are right it is a superfluous flag. Requested changes pushed.
